src/sys/ioctl/mod.rs

```diff
@@ -325,7 +325,7 @@ macro_rules! ioctl {
         pub unsafe fn $name(fd: $crate::libc::c_int,
                             data: $crate::libc::c_int)
                             -> $crate::Result<$crate::libc::c_int> {
-            convert_ioctl_res!($crate::libc::ioctl(fd, iow!($ioty, $nr, ::std::mem::size_of::<$crate::libc::c_int>()) as $crate::sys::ioctl::ioctl_num_type, data))
+            convert_ioctl_res!($crate::libc::ioctl(fd, iowint!($ioty, $nr, ::std::mem::size_of::<$crate::libc::c_int>()) as $crate::sys::ioctl::ioctl_num_type, data))
```

Comment:
Let's remove the ioctl changes from this PR, please comment in #833 with a brief description and some test cases that fail on FreeBSD (see `test/sys/test_ioctl.rs` for existing test cases and please confirm which ones fail). @asomers could you help out here as well?